datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -752,8 +752,8 @@ impl LogicalPlan {
                 }).collect::<Result<Vec<(Expr, Expr)>>>()?;
 
                 Ok(LogicalPlan::Join(Join {
-                    left: Arc::new(inputs[0].clone()),
-                    right: Arc::new(inputs[1].clone()),
+                    left: Arc::new(inputs.swap_remove(0)),
```

Comment:
I think this is correct, but relies on there being exactly 2 inputs which seems like a reasonable assumption at this point